pandas/core/reshape.py

```diff
@@ -1250,40 +1290,84 @@ def get_empty_Frame(data, sparse):
         sparse_series = {}
         N = len(data)
         sp_indices = [[] for _ in range(len(dummy_cols))]
-        for ndx, code in enumerate(codes):
-            if code == -1:
-                # Blank entries if not dummy_na and code == -1, #GH4446
-                continue
-            sp_indices[code].append(ndx)
+        for ndx, code, missing in zip(
```

Comment:
this whole loop is kind of a bummer. I haven't really looked at this code, but would be nice to have a simple vectorized way to fill these codes. This seems like a categorical type operation. @TomAugspurger ?